src/main/java/de/craftlancer/clfeatures/replicator/ReplicatorFeatureInstance.java

```diff
@@ -0,0 +1,333 @@
+        
+        return manager;
+    }
+    
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
+    public void onDaylightDetectInteract(PlayerInteractEvent event) {
+        if (event.getAction() != Action.RIGHT_CLICK_BLOCK)
+            return;
+        
+        Block block = event.getClickedBlock();
+        Player player = event.getPlayer();
+        
+        if (!block.getLocation().equals(daylightCensor))
+            return;
+        
+        event.setCancelled(true);
+        
+        player.getWorld().playSound(player.getLocation(), Sound.BLOCK_SCAFFOLDING_BREAK, 0.5F, 1F);
+        player.openWorkbench(daylightCensor, true);
+    }
+    
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
+    public void onRecipeSet(CraftItemEvent event) {
+        CraftingInventory inventory = event.getInventory();
+        
+        if (inventory.getLocation() == null || !inventory.getLocation().equals(daylightCensor))
+            return;
+        
+        Player player = (Player) event.getWhoClicked();
+        List<Material> blockedProducts = ((ReplicatorFeature) CLFeatures.getInstance().getFeature("replicator")).getBlockedProducts();
+        if (blockedProducts.contains(inventory.getResult().getType())) {
+            player.sendMessage(CLFeatures.CC_PREFIX + ChatColor.YELLOW + "This item is too powerful for the replicator and cannot be crafted.");
+            player.getWorld().playSound(player.getLocation(), Sound.ITEM_SHIELD_BLOCK, 0.5F, 1F);
+            event.setCancelled(true);
+            player.closeInventory();
+            return;
+        }
+        
+        recipe.clear();
+        
+        for (ItemStack item : inventory.getMatrix()) {
+            if (item == null)
+                continue;
+            if (recipe.containsKey(item.getType()))
+                recipe.put(item.getType(), recipe.get(item.getType()) + 1);
+            else
+                recipe.put(item.getType(), 1);
+        }
+        
+        product = inventory.getResult();
+        
+        player.closeInventory();
+        
+        player.getWorld().playSound(player.getLocation(), Sound.BLOCK_SCAFFOLDING_PLACE, 0.5F, 1F);
+        player.sendMessage(CLFeatures.CC_PREFIX + "§eYou have set the replicator to craft §6" + product.getType().name() + "§e.");
+        
+        spawnProduct();
+    }
+    
+    private void spawnProduct() {
+        World world = daylightCensor.getWorld();
+        Location location = daylightCensor.clone();
+        location.setX(location.getX() + 0.5);
+        location.setZ(location.getZ() + 0.5);
+        
+        location.setY(location.getY() + 0.6);
+        
+        if (spawnedItem != null)
+            spawnedItem.remove();
+        
+        spawnedItem = world.dropItem(location, getSingleProduct());
+        spawnedItem.setItemStack(getSingleProduct());
+        spawnedItem.setVelocity(new Vector().zero());
+    }
```

Comment:
Not sure your spawned item handling is chunk unloading and dupe/theft safe. E.g. hoppers, pistons, entities, etc.